nova/virt/vmwareapi/vmops.py

```diff
@@ -1517,18 +1517,24 @@ def _destroy_instance(self, context, instance, destroy_disks=True):
                                            "UnregisterVM", vm_ref)
                 LOG.debug("Unregistered the VM", instance=instance)
 
-                with lockutils.lock(server_group_info.uuid,
-                                lock_file_prefix='nova-vmware-server-group'):
-                    cluster_config = cluster_util.get_cluster_property(
-                        self._session, "configurationEx", self._cluster)
-                    # Check if the unregistered vm is the last vm in its group
-                    for group in cluster_config.group:
-                        if group.name == server_group_info.uuid:
-                            if not hasattr(group, 'vm'):
-                                cluster_util.delete_vm_group(
-                                    self._session, self._cluster,
-                                    group)
-                            break
+                if server_group_info:
+                    with lockutils.lock(server_group_info.uuid,
+                                    lock_file_prefix='vmware-vm-group-policy'):
+                        cluster_config = cluster_util.get_cluster_property(
+                            self._session, "configurationEx", self._cluster)
+                        # Check if the unregistered vm is the last vm in its
+                        # group
+                        for group in cluster_config.group:
+                            if group.name == server_group_info.uuid:
+                                if not hasattr(group, 'vm'):
+                                    LOG.debug("Deleting VM group %s" %
+                                              server_group_info.uuid)
+                                    cluster_util.delete_vm_group(
+                                        self._session, self._cluster,
+                                        group)
+                                    LOG.debug("VM group %s deleted"
+                                              " successfully" % group.name)
+                                break
 
             except Exception as excep:
                 LOG.warning("In vmwareapi:vmops:_destroy_instance, got "
```

Comment:
This log-message is not correct anymore, if the server-group deletion fails. Please use an own try-except for the new code.